NeoML/src/Dnn/DnnDistributed.cpp

```diff
@@ -499,4 +520,100 @@ void CDistributedTraining::StoreDnn( CArchive& archive, int index, bool storeSol
 	}
 }
 
+//---------------------------------------------------------------------------------------------------------------------
+
+void CDistributedInference::initialize( int threadsCount )
+{
+	initThreadGroupInfo();
+
+	// Create reference dnns
+	// To create a reference dnn the original network should be trained or at least reshaped
+	// All training paramBlobs should exist
+	threadParams.Dnns.SetBufferSize( threadsCount );
+	for( int i = 1; i < threadsCount; ++i ) {
+		threadParams.Dnns.Add( referenceDnnFactory.CreateReferenceDnn().Detach() );
+	}
+	// Here it can be either a one more reference dnn
+	// Or also the original dnn, because no one can create a new reference dnn, while this inference
+	threadParams.Dnns.Add( reinterpret_cast<CDnn*>( &referenceDnnFactory ) );
```

Comment:
Oh, please... Please, no... NO... (
It is way too dangerous and probably saves almost nothing in comparison to huge LLMs own runtime. 
The least you can do is just make CDistributedInference friend to CReferenceFactory.

Reply:
make `friend`, fixed